### PR TITLE
SLING-8644: Improve Startorder handling during merge

### PR DESCRIPTION
**Pull request by karlpauls**

### CODE REVIEW

**Review by cziegeler — APPROVED** [review]
Lgtm, thanks

File: src/main/java/org/apache/sling/feature/builder/BuilderUtil.java [hunk 1]
```diff
@@ -131,21 +130,22 @@ static void mergeVariables(Map<String,String> target, Map<String,String> source,
...
+        final Artifacts source,
         final Feature sourceFeature,
             final List<ArtifactId> artifactOverrides,
         final String originKey) {
 
         for (final Artifact artifactFromSource : source) {
 
             // set of artifacts in target, matching the artifact from source
-            final Set<Artifact> allExistingInTarget = new HashSet<>();
+            final Set<Artifact> allExistingInTarget = new LinkedHashSet<>();
```
Comment (bosschaert) [inline src/main/java/org/apache/sling/feature/builder/BuilderUtil.java:143]: Would be good to add a comment to explain why a LinkedHashSet is used?

File: src/main/java/org/apache/sling/feature/builder/BuilderUtil.java [hunk 2]
```diff
@@ -249,6 +255,29 @@ static void mergeBundles(final Bundles target,
                 fromTarget + " and " + fromSource + ". The rule must be specified for " + commonPrefixes);
     }
 
+    private static Artifact selectStartOder(Artifact a, Artifact b, Artifact target) {
```
Comment (bosschaert) [inline src/main/java/org/apache/sling/feature/builder/BuilderUtil.java]: 'Oder' is a typo, oder?

Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/builder/BuilderUtil.java]: yup

File: src/main/java/org/apache/sling/feature/builder/BuilderUtil.java (hunk 1 above)
Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/builder/BuilderUtil.java:143]: I added some comments

**Comment by karlpauls** [comment]
@bosschaert - I think i address your comments. PTAL again please :-)

**Comment by bosschaert** [comment]
LGTM! Thanks @karlpauls !

**Review by bosschaert — APPROVED** [review]

**Pull request merged**